zkvm/src/types.rs

```diff
@@ -213,7 +222,34 @@ impl Predicate {
 
 impl PredicateWitness {
     pub fn to_point(&self) -> CompressedRistretto {
-        unimplemented!()
+        self.to_commitment().compress()
+    }
+
+    fn to_commitment(&self) -> RistrettoPoint {
```

Comment:
to avoid confusion with pedersen commitments, i'd call this method `to_uncompressed_point()`.

Reply:
Also, it seems like `predicate.rs` is better place for this definition, where we could also factor out all the hashing in a single private function.